code/cycler/src/wattrex_battery_cycler/mid/mid_dabs/mid_dabs.py

```diff
@@ -0,0 +1,329 @@
+from __future__ import annotations
+from typing import List #, Dict
+#######################         GENERIC IMPORTS          #######################
+
+#######################       THIRD PARTY IMPORTS        #######################
+
+from system_logger_tool import SysLogLoggerC, sys_log_logger_get_module_logger, Logger
+if __name__ == '__main__':
+    cycler_logger = SysLogLoggerC(file_log_levels= 'log_config.yaml')
+log: Logger = sys_log_logger_get_module_logger(__name__)
+
+# from scpi_sniffer       import DrvScpiHandlerC
+from wattrex_driver_epc import DrvEpcDeviceC, DrvEpcDataC
+# from wattrex_driver_ea  import DrvEaDeviceC, DrvEaDataC
+# from wattrex_driver_rs  import DrvRsDeviceC, DrvRsDataC
+# from wattrex_driver_bk import DrvBkDeviceC, DrvBkDataC
+from wattrex_driver_bms import DrvBmsDeviceC
+from wattrex_battery_cycler_datatypes.cycler_data import (CyclerDataDeviceTypeE, CyclerDataDeviceC,
+                                CyclerDataPwrLimitE, CyclerDataDeviceStatusC, CyclerDataExtMeasC,
+                                CyclerDataGenMeasC, CyclerDataAllStatusC)
+
+#######################          PROJECT IMPORTS         #######################
+
+#######################          MODULE IMPORTS          #######################
+
+#######################              ENUMS               #######################
+
+#######################             CLASSES              #######################
+# TODO: SET PERIODIC TO RECEIVE ELECT AND TEMP MEASURES
+_PERIOD_ELECT_MEAS   = 25 # value *10ms
+_PERIOD_TEMP_MEAS    = 25 # value *10ms
+
+class MidDabsExtraMeterC:
+    """Instanciates an objects that are only able to measures.
+    """
+    def __init__(self, device: CyclerDataDeviceC) -> None:
+        ## TODO: Añadir o no a all status para monitorizar el estado
+        ## Device id will be needed for status device monitoring
+        self.__dev_id: List = device.dev_id
+        self.__device    :  DrvBmsDeviceC |None = None # DrvBkDeviceC |
+        self.__mapping_attr = device.mapping_names
+        if device.device_type is CyclerDataDeviceTypeE.BMS:
+            self.__device : DrvBmsDeviceC = DrvBmsDeviceC(dev_id= device.dev_id,
+                                    can_id= device.iface_name)
+        # elif device.device_type is CyclerDataDeviceTypeE.BK:
+        #     self.__device : DrvBkDeviceC = DrvBkDeviceC(
+        #                                       DrvScpiHandlerC(device.link_conf.__dict__))
+
+    def update(self, ext_meas: CyclerDataExtMeasC) -> None:
+        """Update the external measurements from bms or bk data.
+
+        Args:
+            ext_meas (CyclerDataExtMeasC): [description]
+        """
+        res = None
+        res= self.__device.get_data()
+        # status.extra_meter = CyclerDataDeviceStatusC(error= res.status.error_code,
+        #                                         dev_id= dev_id)
+        if self.__mapping_attr is not None:
```

Comment:
Initialize to empty dict and avoid if case

Reply:
Done